packages/compass-indexes/src/components/regular-indexes-table/regular-indexes-table.tsx

```diff
@@ -140,12 +189,114 @@ const COLUMNS_WITH_ACTIONS: LGColumnDef<IndexInfo>[] = [
   },
 ];
 
+// compassIndexType because type is taken by RegularIndex. indexType is taken by AtlasIndexStats.
+type MappedRegularIndex = RegularIndex & { compassIndexType: 'regular-index' };
+type MappedInProgressIndex = InProgressIndex & {
+  compassIndexType: 'in-progress-index';
+};
+type MappedRollingIndex = RollingIndex & { compassIndexType: 'rolling-index' };
+
+type MergedIndex =
+  | MappedRegularIndex
+  | MappedInProgressIndex
+  | MappedRollingIndex;
+
+function mergeIndexes(
+  indexes: RegularIndex[],
+  inProgressIndexes: InProgressIndex[],
+  rollingIndexes: RollingIndex[]
+): MergedIndex[] {
+  const mappedIndexes: MappedRegularIndex[] = indexes.map((index) => {
+    return { ...index, compassIndexType: 'regular-index' };
+  });
+
+  const mappedInProgressIndexes: MappedInProgressIndex[] =
+    inProgressIndexes.map((index) => {
+      return { ...index, compassIndexType: 'in-progress-index' };
+    });
+
+  const mappedRollingIndexes: MappedRollingIndex[] = rollingIndexes.map(
+    (index) => {
+      return { ...index, compassIndexType: 'rolling-index' };
+    }
+  );
+
+  return [
+    ...mappedIndexes,
+    ...mappedInProgressIndexes,
+    ...mappedRollingIndexes,
+  ];
+}
+
+type CommonIndexInfo = Omit<IndexInfo, 'actions' | 'renderExpandedContent'>;
+
+function getInProgressIndexInfo(index: MappedInProgressIndex): CommonIndexInfo {
+  return {
+    id: index.id,
+    name: index.name,
+    indexInfo: index,
+    type: <TypeField type="unknown" />,
+    size: <SizeField size={0} relativeSize={0} />,
+    usageCount: <UsageField usage={undefined} since={undefined} />,
+    properties: (
+      <PropertyField
+        status={index.status}
+        error={index.error}
+        properties={[]}
+      />
+    ),
+  };
+}
+
+function getRollingIndexInfo(index: MappedRollingIndex): CommonIndexInfo {
+  return {
+    id: `rollingIndex-${index.indexName}`,
+    name: index.indexName,
+    indexInfo: index,
+    // TODO: check that label really is a regular index type
+    type: (
+      <TypeField
+        type={index.indexType.label as RegularIndex['type']}
```

Comment:
Would be nice if the type could already be narrowed from string to a regular index type somehow.